[Fix #12297] Fix a false negative in `Layout/IndentationWidth` for method chain blocks

rscq opened the pull request:
## Summary

Fix a false negative in `Layout/IndentationWidth` when checking indentation inside `do...end` blocks that are part of multiline method chains.

Fixes #12297

## Problem

`Layout/IndentationWidth` did not detect incorrect indentation inside blocks when the block was part of a multiline method chain. This was because the cop used the `end` keyword position as the base for indentation checking, but in method chains, `end` aligns with the method chain start (the dot), not the `do` keyword.

## Solution

Use the dot position (`.method`) as the base for indentation checking when the block is part of a multiline method chain.

## Example

```ruby
# Before: No offense detected (false negative)
foo
  .bar do |x|
  x  # should be indented 2 more spaces
end

# After: Offense detected
foo
  .bar do |x|
  x  # Layout/IndentationWidth: Use 2 (not 0) spaces for indentation.
end

# Correct code
foo
  .bar do |x|
    x
  end
```

## Note

Unrelated style fixes in `unused_block_argument_spec.rb`, `runner_formatter_invocation_spec.rb`,  `ordered_gems.rb` and `ordered_dependencies.rb` were included to pass CI.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

koic commented on lib/rubocop/cop/gemspec/ordered_dependencies.rb:
```diff
@@ -71,12 +71,12 @@ def on_new_investigation
-            register_offense(previous, current)
-          end
+              register_offense(previous, current)
+            end
```
Ditto.

```ruby
           dependency_declarations(processed_source.ast).each_cons(2) do |previous, current|
             next unless consecutive_lines?(previous, current)
             next unless case_insensitive_out_of_order?(gem_name(current), gem_name(previous))
             next unless get_dependency_name(previous) == get_dependency_name(current)

             register_offense(previous, current)
           end
```

koic commented on lib/rubocop/cop/layout/indentation_width.rb:
```diff
@@ -383,6 +386,16 @@ def leftmost_modifier_of(node)
+          send_node = node.send_node
+          receiver = send_node.receiver
+          return false unless receiver&.type?(:send, :block)
+          return false unless send_node.loc?(:dot) && send_node.loc.dot
```
It looks like the tests pass even without this condition. Is it necessary, or is a test case missing?

```suggestion
```

rscq commented:
Thank you for the review, @koic!

I've addressed all of your feedback:

1. **Style fixes** for `ordered_gems.rb` and `ordered_dependencies.rb`: Merged the method chain into a single line to reduce indentation.
2. **Node type coverage**: Changed `type?(:send, :block)` to `type?(:call, :any_block)` to properly handle `:csend` (safe navigation), `:numblock`, and `:itblock`.
3. **`loc?(:dot)` guard**: This guard is required to safely call `.line` on `send_node.loc.dot` in the next line. Added a test case for non-chained blocks (`foo bar do`) to verify this.
4. **Additional test cases**:
    - Safe navigation method chain (`:csend`)
    - Block as receiver (`:block`)
    - Numbered block parameters (`:numblock`)
    - Non-chained block (to test the `loc?(:dot)` guard)

koic commented on spec/rubocop/cop/layout/indentation_width_spec.rb:
```diff
@@ -1659,6 +1659,78 @@ def baz
+            ^{} Use 2 (not 0) spaces for indentation.
+            end
+        RUBY
+      end
```
Can you add a test for `it` block in method chain?
